### PR TITLE
Migrate `client` module to new test structure based on JUnit 5 

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR begins changes required to resolve https://github.com/SpineEventEngine/core-java/issues/701.

All `client` module tests are migrated to the new structure which takes advantage of JUnit 5 capabilities such as [`@DisplayName`](https://junit.org/junit5/docs/5.0.3/api/org/junit/jupiter/api/DisplayName.html) and [`@Nested`](https://junit.org/junit5/docs/5.0.3/api/org/junit/jupiter/api/Nested.html) classes.

`Base` dependency advances to `0.10.40-SNAPSHOT` and so does the Spine version.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: client/src/test/java/io/spine/client/ColumnFiltersTest.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+        private void checkCreatesInstance(CompositeColumnFilter filter,
+                                          CompositeOperator operator,
+                                          ColumnFilter[] groupedFilters) {
+            assertEquals(operator, filter.getOperator());
+            assertContainsAll(filter.getFilterList(), groupedFilters);
+        }
+    }
+
+    @SuppressWarnings("InnerClassMayBeStatic") // JUnit 5 Nested classes cannot to be static.
```
Comment (armiol) [inline client/src/test/java/io/spine/client/ColumnFiltersTest.java]: May we suppress this just once for the whole file?

File: client/src/test/java/io/spine/client/QueryFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author Alex Tymchenko
+ */
+@SuppressWarnings({"LocalVariableNamingConvention",
+        "DuplicateStringLiteralInspection"}) // A lot of similar test display names.
```
Comment (armiol) [inline client/src/test/java/io/spine/client/QueryFactoryTest.java]: Please align.

File: client/src/test/java/io/spine/client/QueryFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+@DisplayName("Query factory should")
+class QueryFactoryTest extends ActorRequestFactoryTest {
+
+    // See {@code client_requests} for declaration.
```
Comment (armiol) [inline client/src/test/java/io/spine/client/QueryFactoryTest.java]: Please have `/** .. */` comment and `{@link ..}` pointing to the declaration.

File: client/src/test/java/io/spine/client/QueryFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+                                                .all(TARGET_ENTITY_CLASS);
+            assertNotNull(readAllQuery);
+
+            checkTypeCorrectAndFiltersEmpty(TARGET_ENTITY_CLASS, readAllQuery);
+
+            checkFieldMaskEmpty(readAllQuery);
+        }
+
+        @Test
+        @DisplayName("create proper `read by ids` query")
```
Comment (armiol) [inline client/src/test/java/io/spine/client/QueryFactoryTest.java]: Can you please change `ids` to `IDs` in display names? That would read much more consistent.

File: client/src/test/java/io/spine/core/CommandsTest.java [hunk 5]
```diff
@@ -117,115 +142,129 @@ public void check_if_contexts_have_same_actor_and_tenantId() {
...
+            final Command command = requestFactory.command()
+                                                  .create(BoolValue.getDefaultInstance());
+            assertTrue(Commands.wereAfter(secondsAgo(5))
+                               .apply(command));
+        }
+
+        @Test
+        @DisplayName("support `wereBetween` predicate")
+        void createWereBetweenPredicate() {
+            final Command command1 = requestFactory.createCommand(StringValue.getDefaultInstance(),
```
Comment (armiol) [inline client/src/test/java/io/spine/core/CommandsTest.java]: Let's use this moment and rename these local declarations to `fiveMinsAgo`, `twoMinsAgo` etc.

File: client/src/test/java/io/spine/grpc/StreamObserversTest.java [hunk 6]
```diff
@@ -143,46 +152,52 @@ private static void checkIsCompleted(MemoizingObserver<Object> observer) {
...
+        @DisplayName("return Error extracted from StatusException metadata")
+        void processStatusException() {
+            final Error expectedError = Error.getDefaultInstance();
+            final Metadata metadata = MetadataConverter.toMetadata(expectedError);
+            final StatusException statusException = INVALID_ARGUMENT.asException(metadata);
 
-        assertEquals(expectedError, StreamObservers.fromStreamError(statusException)
-                                                   .get());
-    }
+            assertEquals(expectedError, StreamObservers.fromStreamError(statusException)
```
Comment (armiol) [inline client/src/test/java/io/spine/grpc/StreamObserversTest.java]: Please use a static import for `StreamObservers.fromStreamError(...)`.

File: testutil-client/src/main/java/io/spine/client/CommonTestNames.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * <p>This class can be used to avoid string literal duplication when assigning
+ * {@link org.junit.jupiter.api.DisplayName} to the common {@linkplain org.junit.jupiter.api.Test
+ * test cases}.
+ *
+ * @author Dmytro Kuzmin
+ */
+@Internal
+@VisibleForTesting
+public class CommonTestNames {
```
Comment (armiol) [inline testutil-client/src/main/java/io/spine/client/CommonTestNames.java]: This is `DisplayNames`.

And I would turn that into an enumeration.

File: testutil-client/src/main/java/io/spine/client/CommonTestNames.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * A name for the test cases checking that a class has private parameterless (aka "utility")
+     * constructor.
+     */
+    public static final String UTILITY_CTOR = "have private parameterless constructor";
```
Comment (armiol) [inline testutil-client/src/main/java/io/spine/client/CommonTestNames.java]: This is `HAVE_PARAMETERLESS_CTOR` — in order to read this easier when used in `@DisplayName`.

File: testutil-client/src/main/java/io/spine/client/CommonTestNames.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * A name for test cases checking that class public methods do not accept {@code null} for
+     * their non-{@linkplain javax.annotation.Nullable nullable} arguments.
+     */
+    public static final String NULL_TOLERANCE =
```
Comment (armiol) [inline testutil-client/src/main/java/io/spine/client/CommonTestNames.java]: For the same reason as above, this is `NOT_ACCEPT_NULLS`.

File: testutil-client/src/main/java/io/spine/client/CommonTestNames.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            "not accept nulls for non-Nullable public method arguments";
```
Comment (armiol) [inline testutil-client/src/main/java/io/spine/client/CommonTestNames.java]: BTW, we also test package-private and protected methods for zero null-tolerance. So the caption and its documentation don't always tell the true.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.
